### ISSUE TITLE
Suggestion feature

### ISSUE THREAD

**Issue by jreason** [BobKnothe/autoNumeric#222]
Hi,

You have the option to pad the decimal places.
You also have the option to allow leading zeros.

It would be good if you could also have the option to pad none decimals?
So for example, the number must always be a 3 digit number.
If 1 was entered, it would pad to 001
If 50 was entered, it would pad to 050

Cheers
James

**Comment by BobKnothe** [comment]
James,

Thanks for the suggestion - FYI You are the first to ask for this feature. 

May I ask the purpose, what is the underlining need? 

Bob

**Comment by jreason** [comment]
Hi Bob,

I’m dealing with some European postcodes.

Spain for example is always has to be 5 digits.
So area code 10, would need to be 00010.

When the fields are being edited, if a user only types 10, it should be auto padded to avoid breaking the required format.

Cheers
James

**Comment by jlaustill** [comment]
I could actually use this feature as well.  We use account numbers that
have the same issue, must always be 16 digits, even if that is
0000000000000001, and I really struggle with it everytime.

Interesting suggestion for sure :)
Joshua

On Tue, Jul 12, 2016 at 9:00 AM, jreason <notifications@github.com> wrote:

> Hi Bob,
>
> I’m dealing with some European postcodes.
>
> Spain for example is always has to be 5 digits.
> So area code 10, would need to be 00010.
>
> When the fields are being edited, if a user only types 10, it should be
> auto padded to avoid breaking the required format.
>
> Cheers
> James
>
>
>
> From: Bob Knothe [mailto:notifications@github.com]
> Sent: 12 July 2016 15:55
> To: BobKnothe/autoNumeric <autoNumeric@noreply.github.com>
> Cc: jreason <reasy_@hotmail.com>; Author <author@noreply.github.com>
> Subject: Re: [BobKnothe/autoNumeric] Suggestion feature (#222)
>
>
> James,
>
> Thanks for the suggestion - FYI You are the first to ask for this feature.
>
> May I ask the purpose, what is the underlining need?
>
> Bob
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub<
> https://github.com/BobKnothe/autoNumeric/issues/222#issuecomment-232073498>,
> or mute the thread<
> https://github.com/notifications/unsubscribe/AKvRFpN7nM0gMzqLmfScQ9KJIHICFYCgks5qU6rYgaJpZM4JJYj5>.
>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BobKnothe/autoNumeric/issues/222#issuecomment-232075381>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AKTdh3V2VTPmwZaE5YQEJu8QOmWb4H75ks5qU6wkgaJpZM4JJYj5>
> .
>



-- 
{
   name: "Joshua Austill",
   status: "ttyl"
}

**Comment by BobKnothe** [comment]
Joshua,

I understand and see this feature being useful as a mask. In an account number I could even see leading and trailing padding being supported at the same time.

The underling basis of AN is formatting currency and numeric values and this feature as you envision deviates  from that core.

This will take some take time to consider because this will effect many of the supported methods.

Bob

**Comment by jlaustill** [comment]
Fair enough. Not all account numbers actually are numeric either.  I see
that as more of an edge case than a standard.

Joshua

On Tuesday, July 12, 2016, Bob Knothe <notifications@github.com> wrote:

> Joshua,
>
> I understand and see this feature being useful as a mask. In an account
> number I could even see leading and trailing padding being supported at the
> same time.
>
> The underling basis of AN is formatting currency and numeric values and
> this feature as you envision deviates from that core.
>
> This will take some take time to consider because this will effect many of
> the supported methods.
>
> Bob
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/BobKnothe/autoNumeric/issues/222#issuecomment-232117015>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AKTdh9TjBB5_d_NUCC9DiasO-2GEB8S5ks5qU83EgaJpZM4JJYj5>
> .
>


-- 
{
   name: "Joshua Austill",
   status: "ttyl"
}

**Comment by kazim1510** [comment]
Hi,
I need this feature as well
